activitysim/abm/models/util/tour_frequency.py

```diff
@@ -435,3 +435,240 @@ def process_joint_tours(joint_tour_frequency, joint_tour_frequency_alts, point_p
+        tours.tour_num        - index of tour (of any type) for parent
+        tours.tour_count      - number of tours of any type) for parent (parent's max tour_num)
+        tours.tour_category   - one of 'mandatory', 'non_mandatory', 'atwork', or 'joint'
+    """
+
+    # get the actual alternatives for each person - have to go back to the
+    # non_mandatory_tour_frequency_alts dataframe to get this - the choice
+    # above just stored the index values for the chosen alts
+    tour_counts = joint_tour_frequency_composition_alts.loc[joint_tour_frequency_composition]
+
+    # assign person ids to the index
+    tour_counts.index = joint_tour_frequency_composition.index
+
+    """
+               alt1       alt2     alt3
+    <parent_col>
+    2588676       2         0         0
+    2588677       1         1         0
+    """
+
+    tours = create_joint_tours(tour_counts, tour_category, parent_col)
+
+    return tours
+
+def create_joint_tours(tour_counts, tour_category, parent_col='person_id'):
+    """
+    This method processes the tour_frequency column that comes
+    out of the model of the same name and turns into a DataFrame that
+    represents the tours that were generated
+    
+    Parameters
+    ----------
+    tour_counts: DataFrame
+        table specifying how many tours of each type to create
+        one row per person (or parent_tour for atwork subtours)
+        one (int) column per tour_type, with number of tours to create
+    tour_category : str
+        one of 'mandatory', 'non_mandatory', 'atwork', or 'joint'
+    
+    Returns
+    -------
+    tours : pandas.DataFrame
+        An example of a tours DataFrame is supplied as a comment in the
+        source code - it has an index which is a unique tour identifier,
+        a person_id column, and a tour type column which comes from the
+        column names of the alternatives DataFrame supplied above.
+        
+        tours.tour_type       - tour type (e.g. school, work, shopping, eat)
+        tours.composition     - tour composition (e.g. adults, children, mixed)
+        tours.tour_type_num   - if there are two 'school' type tours, they will be numbered 1 and 2
+        tours.tour_type_count - number of tours of tour_type parent has (parent's max tour_type_num)
+        tours.tour_num        - index of tour (of any type) for parent
+        tours.tour_count      - number of tours of any type) for parent (parent's max tour_num)
+        tours.tour_category   - one of 'mandatory', 'non_mandatory', 'atwork', or 'joint'
+    """
+
+    # FIXME - document requirement to ensure adjacent tour_type_nums in tour_num order
+
+    """
+               alt1       alt2     alt3
+    <parent_col>
+    2588676       2         0         0
+    2588677       1         1         0
+    """
+    tour_type_dict={5:'shopping',6:'othmaint',7:'eatout',8:'social',9:'othdiscr'}
+    tour_comp_dict={1:'adults',2:'children',3:'mixed'}
```

Comment:
Can we expose these settings in the model yaml?  That way the user can change the tour purposes.

Reply:
Good suggestion. Done.